compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3281,6 +3282,80 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn suggest_bounds_for_range_to_method(
+        &self,
+        err: &mut Diag<'_>,
+        source: SelfSource<'tcx>,
+        item_ident: Ident,
+    ) {
+        if let SelfSource::MethodCall(rcvr_expr) = source
+            && let hir::ExprKind::Struct(qpath, fields, _) = rcvr_expr.kind
+            && let hir::QPath::LangItem(lang_item, _) = qpath
+        {
+            let is_inclusive = match lang_item {
+                hir::LangItem::RangeTo => false,
+                hir::LangItem::RangeToInclusive | hir::LangItem::RangeInclusiveCopy => true,
+                _ => return,
+            };
+
+            let tcx = self.tcx;
+            if let Some(iterator_trait) = tcx.get_diagnostic_item(sym::Iterator) {
+                let has_method = tcx
+                    .associated_items(iterator_trait)
+                    .filter_by_name_unhygienic(item_ident.name)
+                    .next()
+                    .is_some();
+
+                if !has_method {
+                    return;
+                }
+            }
+
+            let element_ty = self.typeck_results.borrow().expr_ty_opt(rcvr_expr).and_then(|ty| {
+                if let ty::Adt(_, args) = ty.kind() {
+                    args.get(0).and_then(|arg| arg.as_type())
+                } else {
+                    None
+                }
+            });
+
+            let is_integral = element_ty.is_some_and(|ty| ty.is_integral());
+            let source_map = self.tcx.sess.source_map();
+            let range_type = if is_inclusive { "RangeInclusive" } else { "Range" };
+
+            // Check if the end bound is a negative number
+            let end_is_negative = is_integral
+                && fields.iter().find(|f| f.ident.name == rustc_span::sym::end).is_some_and(
+                    |end_field| {
+                        matches!(end_field.expr.kind, hir::ExprKind::Unary(rustc_ast::UnOp::Neg, _))
```

Comment:
While this could also be `-x` or `--1`, suggesting `/* start */` instead of `0` is probably for the better :+1: